File: Module4/11_WebService_RESTful/thuc_hanh/cutomer_management/src/main/java/com/codegym/service/CustomerService.java
package com.codegym.service;

import com.codegym.model.Customer;
import org.springframework.stereotype.Service;

import java.util.List;
@Service
public interface CustomerService {
    List<Customer> findALL();

    Customer findById(Long id);

    void save (Customer customer);

    void remove (Long id);
}

File: Module5/10_CaseStudy/case-study/src/app/customer/customer.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {ICustomer} from "./model/customer";

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  private API_URL = 'http://localhost:3000/customerList';
  message! : String;

  constructor(private httpClient : HttpClient) {
    console.log('service constructor');
  }
  findAll(): Observable<ICustomer[]>{
    return this.httpClient.get<ICustomer[]>(this.API_URL);
  }
  save(customer:ICustomer): Observable<void>{
    return  this.httpClient.post<void>(this.API_URL, customer)
  }
}
